src/main/java/sparksoniq/spark/ml/ApplyTransformerRuntimeIterator.java

```diff
@@ -97,6 +97,16 @@ public ApplyTransformerRuntimeIterator(
 
             return result;
         } catch (IllegalArgumentException e) {
+            if (e.getMessage().matches(".*DecimalType.*is not supported.*")) {
```

Comment:
Note for the future: promotion is more general than mapping decimals to doubles (e.g., integer to decimal, anything to double). Maybe opening an issue to remember for later could be useful if you believe not all cases of the spec are covered.